data/trie/sync_test.go

```diff
@@ -43,3 +49,75 @@ func TestTrieSync_InterceptedNodeShouldNotBeAddedToNodesForTrieIfNodeReceived(t
 	assert.True(t, ok)
 	assert.Equal(t, bn, nodeInfo.trieNode)
 }
+
+func TestTrieSync_InterceptedNodeTimedOut(t *testing.T) {
+	t.Parallel()
+
+	timeout := time.Second * 2
+	arg := ArgTrieSyncer{
+		RequestHandler:   &mock.RequestHandlerStub{},
+		InterceptedNodes: testscommon.NewCacherMock(),
+		Trie: &patriciaMerkleTrie{
+			trieStorage: &mock.StorageManagerStub{
+				DatabaseCalled: func() data.DBWriteCacher {
+					return mock.NewMemDbMock()
+				},
+			},
+		},
+		ShardId:                        0,
+		Topic:                          "trieNodes",
+		TrieSyncStatistics:             statistics.NewTrieSyncStatistics(),
+		TimeoutBetweenTrieNodesCommits: timeout,
+	}
+	ts, err := NewTrieSyncer(arg)
+	require.Nil(t, err)
+
+	start := time.Now()
+	err = ts.StartSyncing([]byte("roothash"), context.Background())
+	end := time.Now()
+
+	assert.True(t, errors.Is(err, ErrTimeIsOut))
+	assert.True(t, timeout <= end.Sub(start))
+}
+
+func TestTrieSync_FoundInStorageShouldNotRequest(t *testing.T) {
+	t.Parallel()
+
+	timeout := time.Second * 200
+	marsh, hasher := getTestMarshAndHasher()
```

Comment:
marsh sounds weird, also in the method name

Reply:
renamed